docs/proposals/001-k8s-status.md

```diff
@@ -0,0 +1,433 @@
+to debug this is to look for logs in the `k8sd` service, which users might not
+be familiar with.
+
+Instead, we want the `k8s status` command to return meaningful information about
+the current _deployment status_ of each feature. We want to replace the
+following output (taking the `dns` feature as an example):
+
+```bash
+$ k8s status
+...
+dns:
+  enabled: true                         # these fields comes from the cluster
+  service-ip: 10.152.183.10             # configuration. enabled=true does not
+  cluster-domain: cluster.local         # mean that 'dns' is actually available
+```
+
+With something like this:
+
+```bash
+$ k8s status
+...
+dns:
+  enabled: true/false                   # true if successful, false otherwise
+  message: CoreDNS deployed
+```
+
+## User facing changes
+<!--
+This section MUST describe any user-facing changes that this feature brings, if
+any. If an API change is required, the affected endpoints MUST be mentioned. If
+the output of any k8s command changes, the difference MUST be mentioned, with a
+clear example of "before" and "after".
+-->
+
+### `k8s status` command
+
+The output format of `k8s status` will change.
+
+#### Old output
+
+```bash
+$ k8s status
+status: ready
+high-availability: no
+datastore:
+  type: k8s-dqlite
+  voter-nodes:
+    - 10.87.24.119:6400
+  standby-nodes: none
+  spare-nodes: none
+network:
+  enabled: true
+dns:
+  enabled: true
+  cluster-domain: cluster.local
+  service-ip: 10.152.183.104
+  upstream-nameservers:
+  - /etc/resolv.conf
+ingress:
+  enabled: false
+  default-tls-secret: ""
+  enable-proxy-protocol: false
+load-balancer:
+  enabled: false
+  cidrs: []
+  l2-mode: false
+  l2-interfaces: []
+  bgp-mode: false
+  bgp-local-asn: 0
+  bgp-peer-address: ""
+  bgp-peer-asn: 0
+  bgp-peer-port: 0
+local-storage:
+  enabled: true
+  local-path: /var/snap/k8s/common/rawfile-storage
+  reclaim-policy: Delete
+  default: true
+gateway:
+  enabled: true
+```
+
+#### New output
+
+With `--output-format=plain`, a simple overview is shown, with `yaml` or `json`
+more fields are shown:
+
+```bash
+$ k8s status
+ready: true
+status: ready
+high-availability: no
+datastore:
+  type: k8s-dqlite
+  voter-nodes:
+    - 10.87.24.119:6400
+  standby-nodes: none
+  spare-nodes: none
+network:
+  enabled: true
+  message: Cilium is deployed
+dns:
+  enabled: true
+  message: CoreDNS is deployed
+ingress:
+  enabled: false
+  message: ""
+load-balancer:
+  enabled: true
+  message: Cilium configured for LoadBalancer services
+local-storage:
+  enabled: true
+  message: Rawfile LocalPV is deployed
+gateway:
+  enabled: false
+  message: Cilium GatewayAPI support could not be enabled, error was "..."
+```
+
+When the output format is `json` or `yaml`, more fields may be shown:
+
+```bash
+$ k8s status --output-format=yaml
+...
+network:
+  enabled: true
+  message: Cilium is deployed
+  timestamp: 2020-01-01 10:00:00
```

Comment:
Why do we need the timestamp?

Reply:
As a means of explaining _when_ the last update was done. It is different to see a recent timestamp, vs a timestamp 3 months ago

Reply:
If that's the latest update, can we call it `updated:` or sth along those lines?